content/docs/how-we-work/coding-principles-standards.md

```diff
@@ -45,3 +45,7 @@ In our projects, we strictly adhere to the established coding standards. These s
 - We stand behind the YAGNI (You Aren’t Gonna Need It) philosophy and avoid “gold plating”.
 
   - When writing code, our code should solve ONLY and EXACTLY the use case that we are trying to support. In many instances, we have a desire to make things MORE FLEXIBLE in the FUTURE but that is code we will have to support without knowing whether any of those use cases will ever come up. Another big reason for wanting to be FLEXIBLE is that refactoring is hard, and it feels easier to have flexibility at the front, but that is why we write automated tests, so refactoring will not be as painful.
+
+## Pair Programming
+
+Pair programming is a software development technique where two developers work together. Sometimes it is proven helpful to debug code collectively by connecting on calls, sometimes [rubber-ducking]({{< relref "../common-areas/debugging/#rubber-ducking" >}}) tactics are clearly achieved through this approach. Pair programming generally helps in improving code quality, including productivity and apparently communication skills.
```

Comment:
Why "apparently" communication skills?

Reply:
Is it reflecting that we are unsure that it improves communication skill as well? If so, we can be direct there by removing 'apparently'.

Reply:
I don't know if anyone has ever linked communication skills with peer programming. That said, I don't think this is something that adds value. I mean, no one thinks that I need to improve my communication skills, let me start pair programming. It's not a direct outcome, as far as I know.

Reviewing more, I don't think anyone has measured a link between productivity and pair programming. Again, it depends on the meaning of productivity. For example, there is anecdotal evidence that pair programming takes the same amount of time as programming alone but the benefits are in quicker PR reviews (typically zero waiting time because a PR often doesn't need a review anymore after pair programming). However, it may appear that the throughput is reduced which is why some managers frown at the idea (anecdotally). It improves long-term results at the expense of shorter ones and that's hard to see.

The biggest and most relevant benefit of pair programming (and ensemble programming) is that it improves code quality in the first iteration itself, improves the lottery factor (bus factor), increases knowledge sharing, eliminates waiting for code reviews, and results in healthier teams. It's easy to stretch all this to productivity gains and communication skills but I don't think there is a direct link established.

Reply:
Thank you @hussainweb for the review. 

I agree to your point `no one thinks that I need to improve my communication skills, let me start pair programming`. I put this point based on my experience-  I have learned a few terminologies, some improved way of presenting/addressing issues during pair-programming with my seniors, even  references like [geeksforgeeks](https://www.geeksforgeeks.org/pair-programming/#:~:text=Team%20develops%20better%20communication%20skills) do support that it can help in developing better communication skills.

And around productivity, I agree to your points again, even this [Studies section](https://en.wikipedia.org/wiki/Pair_programming#Studies:~:text=On%20simple%20tasks,them.%5B9%5D) too validates it. :)

And for the content, I thought of using details shared in the last paragraph, and came up with:
> Pair programming is a software development technique where two developers work together. Sometimes it is proven helpful to debug code collectively by connecting on calls, sometimes [rubber-ducking]({{< relref "../common-areas/debugging/#rubber-ducking" >}}) tactics are clearly achieved through this approach. The benefit of pair programming is that it improves code quality in the first iteration itself, improves the [lottery factor (bus factor)]({{< relref "../behaviors/team/#understand-lottery-factor" >}}), increases knowledge sharing, eliminates waiting for code reviews, and results in healthier teams.

Please review.